Added login notice to single lesson & single quiz pages.

hansschuijff opened the pull request:
Fixes #3150

### Changes proposed in this Pull Request

* Add login notice to single lesson & single quiz page, when user is not logged and login is required to get access.
* Also moved the add_action for "sensei_single_course_content_inside_before" up in templates.php, since it belongs to the single course hooks and was inserted between the single lesson hooks.

hansschuijff committed:
Added login notice to single lesson page.

hansschuijff committed:
Lint-errors corrected

hansschuijff committed:
Correct @version comment of lesson add_action.

hansschuijff committed:
Added login notice to single quiz too.

hansschuijff committed:
Fixed small error: $id instead of $lesson_id.

hansschuijff committed:
Fixed reported lint error.

hansschuijff committed:
Small docblock change.

hansschuijff committed:
Corrected the comment on login_notice in utils.

hansschuijff commented:
@donnapep I took a swing at issue #3150 and added a login notice to both the single lesson and the single quiz pages. 

I updated the @hooked comments in 2 templates, added some callbacks that probably should have been mentioned there already (or not if you want to stop that practice all together), but didn't know if you want to change versions, since only the comments changed. 

I deleted one statement in the existing code that was there twice and was redundant.

I didn't add filters on the new notices, but if you want them they are added easy enough.

Let me know what you think, when you get around to it. I'll leave this one for now.

donnapep commented:
Thanks for this.

When a user is not logged in, rather than showing them two notices like this:
![Screen Shot 2020-10-14 at 2 06 56 PM](https://user-images.githubusercontent.com/1190420/96028199-ff515b00-0e26-11eb-8612-f6eb595a2a2a.jpg)

It would be preferable to display a single notice:

![Screen Shot 2020-10-14 at 2 12 14 PM](https://user-images.githubusercontent.com/1190420/96028434-46d7e700-0e27-11eb-96ca-aa449e8c5bc5.jpg)

Then, once logged in and if they still haven't started the course, they would only see the original "Please sign up..." message. Ditto for the quiz.

hansschuijff commented:
Hi @donnapep 

I'll take a crack at it and will update the PR. 

In the code there is already a notice for the quiz page with the text "You must be logged in to take this quiz". There I can just switch the if-else to first test for user logged in and it will probably work like you propose.  Do you want to make the notice text the same as your proposal for lesson? So change the text to "Please log in to take this quiz"? 

I also saw that the same logic should probably apply to the module page. There the enrol first notice is issued, but the situation is basically the same. And when a user should be logged in, there should be a notice saying that.

Hans

donnapep commented:
> Do you want to make the notice text the same as your proposal for lesson? So change the text to "Please log in to take this quiz"?

Sounds good. Consistency for the win. 🙂 I think there might even be a period missing on the existing quiz notice.

> I also saw that the same logic should probably apply to the module page. There the enrol first notice is issued, but the situation is basically the same. And when a user should be logged in, there should be a notice saying that.

All makes sense to me. Thanks!

hansschuijff committed:
Applied info box styles also to module pages.

Added .module-container to info-box selectors so
styling of sensei messages on module pages is consistent
with other sensei pages.

hansschuijff committed:
No course signup notice on lesson when logged out.

Don't show course signup notice when user is not logged in.
In that case the user is prompted to log in first.

hansschuijff committed:
Changed Quiz to quiz in course signup notice.

hansschuijff committed:
Don't show other notices when user should login.

- Changed login_notice text to "Please log in to access {lesson|quiz}".
- Used Sensei()->notices->add_notice() for quiz notices too.
- Uses Sensei()->notices->add_notice() also for quiz saved message.
- Probably Sensei()->frontend->messages is now redundant,
  but kept it just in case.

hansschuijff committed:
Added login notice to module pages.

hansschuijff commented:
Hi @donnapep 

I think this is it. I made a uniform login notice for lessons, quizes and modules and it seems to work fine right now. There seems to be a conflict that prevents the merge, but that is probably something you need to look at, since I have no write access. The templates I only addes some comments to, so that can be removed and the scss I added modules to the notice-styles, since before they wheren't shown in the same styling as the notices on other sensei pages.
Let me know if there is anything else...
For now, I'm done with it.

hansschuijff commented:
@donnapep 
The @since in the docblocks I will update when you tell me what version you want there. For now they are set on 3.2.0, because I started this pr when the current version was still 3.1.1.

I think frontend->messages may be redundant after this pr, but that's for you to decide. I only found a save quiz message that was added that way and think it might be a legacy property.

The pr uses Sensei()->notices->add_notice() instead of echo (what was used in quiz). I thought that would probably be the prefered way, since most code seems to use it at this moment.

donnapep commented:
Now that we are starting to move away from templates and towards blocks, I've added thoughts on a [block-based solution](https://github.com/Automattic/sensei/issues/3150#issuecomment-788114089).
